test/objects.test.ts

```diff
@@ -134,6 +134,216 @@ describe("Objects", () => {
     type test6 = Expect<
       Equal<res6, { tuple?: [string?, [string?, number?]?] }>
     >;
+
+    // interface ExtendedFunction extends Function {
+    //   args: 0;
+    // }
+
+    // interface ExtendedDate extends Date {
+    //   today(): number;
+    // }
+
+    // interface ExtendedError extends Error {
+    //   code: string;
+    // }
+
+    // interface ExtendedRegExp extends RegExp {
+    //   caseSensitive: true;
+    // }
+
+    type ComplexNestedRequired = {
+      simple: number;
+      nested: {
+        date: Date;
+        func: () => string;
+        array: { bar: number }[];
+        tuple: [string, number, { good: boolean }];
+        set: Set<{ name: string }>;
+        map: Map<
+          string,
+          {
+            name: string;
+          }
+        >;
+        promise: Promise<{ foo: string; bar: number }>;
+      };
+    };
+
+    type ComplexNestedPartial = {
+      simple?: number;
+      nested?: {
+        date?: Date;
+        func?: () => string;
+        array?: ({ bar?: number } | undefined)[];
+        set?: Set<{ name?: string }>;
+        tuple?: [string?, number?, { good?: boolean }?];
+        map?: Map<
+          string,
+          {
+            name?: string;
+          }
+        >;
+        promise?: Promise<{ foo?: string; bar?: number }>;
+      };
+    };
+
+    type cases = [
+      Expect<Equal<Call<Objects.PartialDeep, number>, number>>,
+      Expect<Equal<Call<Objects.PartialDeep, string>, string>>,
+      Expect<Equal<Call<Objects.PartialDeep, boolean>, boolean>>,
+      Expect<Equal<Call<Objects.PartialDeep, bigint>, bigint>>,
+      Expect<Equal<Call<Objects.PartialDeep, symbol>, symbol>>,
+      Expect<Equal<Call<Objects.PartialDeep, undefined>, undefined>>,
+      Expect<Equal<Call<Objects.PartialDeep, null>, null>>,
+      // Expect<Equal<Call<Objects.PartialDeep, Function>, Function>>,
+      // Expect<
+      //   Equal<
+      //     Call<Objects.PartialDeep, ExtendedFunction>,
+      //     Function & { args: 0 }
+      //   >
+      // >,
+      // Expect<Equal<Call<Objects.PartialDeep, Date>, Date>>,
+      // Expect<
+      //   Equal<
+      //     Call<Objects.PartialDeep, ExtendedDate>,
+      //     Date & { today(): number }
+      //   >
+      // >,
+      // Expect<Equal<Call<Objects.PartialDeep, Error>, Error>>,
+      // Expect<
+      //   Equal<
+      //     Call<Objects.PartialDeep, ExtendedError>,
+      //     Error & { code: string }
+      //   >
+      // >,
+      // Expect<Equal<Call<Objects.PartialDeep, RegExp>, RegExp>>,
+      // Expect<
+      //   Equal<
+      //     Call<Objects.PartialDeep, ExtendedRegExp>,
+      //     RegExp & { caseSensitive: true }
+      //   >
+      // >,
```

Comment:
There are object types that are used differently – Function, Date, Error and RegExp, is there anything that we need to do here? Otherwise I'm happy to remove these comments